### PR TITLE
Feature/allow redis password

### PR DESCRIPTION
**Pull request by verboEse**
Add authentication against Redis with password.

This should solve #472.

Known error: After validation the password field is empty. Maybe you can push me to how I can fill it.

### CODE REVIEW

**Comment by verboEse** [comment]
I just added the missing mail address to my account, @cla-bot should verify now correctly.

**Comment by nilmerg** [comment]
@cla-bot check

**Comment by bobapple** [comment]
@cla-bot check

File: library/Icingadb/Setup/RedisStep.php [hunk 1]
```diff
@@ -117,6 +119,10 @@ public function getSummary()
             new HtmlElement('strong', null, Text::create(t('Port'))),
             $this->data['redis1_port'] ?: 6380
         ]));
+        $primaryOptions->addHtml(Table::row([
+            new HtmlElement('strong', null, Text::create(t('Password'))),
+            $this->data['redis1_password'] ?: 6380
```
Comment (nilmerg) [inline library/Icingadb/Setup/RedisStep.php:124]: Use `mt('icingadb', 'None', 'non-existence of a value')` as default here.

File: library/Icingadb/Setup/RedisStep.php [hunk 2]
```diff
@@ -133,6 +139,10 @@ public function getSummary()
                 new HtmlElement('strong', null, Text::create(t('Port'))),
                 $this->data['redis2_port'] ?: 6380
             ]));
+            $secondaryOptions->addHtml(Table::row([
+                new HtmlElement('strong', null, Text::create(t('Password'))),
+                $this->data['redis2_password'] ?: 6380
```
Comment (nilmerg) [inline library/Icingadb/Setup/RedisStep.php:144]: Use `mt('icingadb', 'None', 'non-existence of a value')` as default here.

**Review by nilmerg — CHANGES_REQUESTED** [review]
> Known error: After validation the password field is empty. Maybe you can push me to how I can fill it.

Define `'renderPassword' => true` as option.

File: library/Icingadb/Setup/RedisStep.php (hunk 2 above)
Comment (verboEse) [reply library/Icingadb/Setup/RedisStep.php]: changed this, too.

**Comment by verboEse** [comment]
I added the 'non existence' part for empty password values, but I still don't see how to re-fill the field after validation. `'renderPassword' => true` makes no difference?!

**Comment by nilmerg** [comment]
Where have you added it? It needs to go to both password elements. It works for me this way fine.

**Comment by verboEse** [comment]
You are right. I got the wrong spot for this. Now it works. Thanks for that hint, I learned ... :-)

**Review by nilmerg — APPROVED** [review]
Thanks!

**Pull request merged**